### DISCUSSION TITLE
Arbitrum support?

### DISCUSSION THREAD

**Discussion by jornx** [uniswap-python/uniswap-python#183] [Q&A]
Hi! I was wondering if you guys were planning to add Arbitrum support?

**Comment by liquid-8** [comment]
From the quick research I've done it seems to me that it should work out of the box. You just have to specify proper endpoint and chainId then here you go.

**Comment by liquid-8** [comment]
https://github.com/uniswap-python/uniswap-python/discussions/176

**Comment by theLockesmith** [comment]
There are a lot of quirks I've run into trying to use this on Arbitrum. The two biggest being that any function that calls `.transact` rather than using `_build_and_send_tx` the calls fail.

Another one seems to be using the `_build_and_send_tx` the transaction will fail estimating gas. That is easily fixed by building my own tx parameters though.

Certainly not one to one from what I can tell, but I've also not tried using this on mainnet to see if it's just something I'm doing wrong or if it's specific to Arbitrum.
